only disallow _local_ users setting usermode -i when feature AUTOINVISIBLE is turned on
and do not enforce this setting on remote users, leads to desynch

diff -r 096116177219 ircd/s_user.c
--- a/ircd/s_user.c
+++ b/ircd/s_user.c
@@ -1252,8 +1252,7 @@
         if (what == MODE_ADD)
           SetInvisible(sptr);
         else
-          if (!feature_bool(FEAT_AUTOINVISIBLE) || IsOper(sptr)) /* Don't allow non-opers to -i if FEAT_AUTOINVISIBLE is set */
-            ClearInvisible(sptr);
+          ClearInvisible(sptr);
         break;
       case 'd':
         if (what == MODE_ADD)
@@ -1368,6 +1367,12 @@
     if (do_set_host && !hostmask && !FlagHas(&setflags, FLAG_LOCOP) && !FlagHas(&setflags, FLAG_OPER))
       do_set_host = 0;
 
+    /* disallow ordinary users to do MODE -i when feature AUTOINVISIBLE is enabled */
+    if (FlagHas(&setflags, FLAG_INVISIBLE) && !IsInvisible(sptr) &&          /* MODE -i */
+        feature_bool(FEAT_AUTOINVISIBLE) &&                                  /* AUTOINVISIBLE is enabled */
+        !FlagHas(&setflags, FLAG_OPER) && !FlagHas(&setflags, FLAG_LOCOP))   /* (was) not opered */
+       SetInvisible(sptr);
+
     /*
      * only send wallops to opers
      */
